test-framework/common/src/main/java/io/quarkus/test/common/QuarkusTestResource.java

```diff
@@ -20,11 +20,13 @@
  *
  * Note that test resources are never restarted when running {@code @Nested} test classes.
  *
+ * @deprecated Use the new {@link WithTestResource} instead.
```

Comment:
We probably want to mention that it will be a long while until we actually remove it.